src/metabase/models/params/field_values.clj

```diff
@@ -46,9 +46,9 @@
   [field-ids]
   (when (seq field-ids)
     (not-empty
-      (let [field-values       (t2/select [FieldValues :values :human_readable_values :field_id]
-                                          :type :full
-                                          :field_id [:in (set field-ids)])
+      (let [field-values       (map #(select-keys (field-values/get-or-create-full-field-values! %)
+                                                  [:field_id :human_readable_values :values])
+                                     (t2/select :model/Field :id [:in (set field-ids)]))
```

Comment:
This will make a DB call to fetch the field values per dashboard parameter with field values. If we use `(t2/hydrate (t2/select :model/Field :id [:in (set field-ids)]) :values)` instead to get the field values for all fields, we will only make one DB call and should save a lot of waiting for round trips in the worst case. We'd need to modify `get-or-create-full-field-values!` to optionally get the field values from the `:values` key.

This should make the worst case faster:
```suggestion
      (let [field-values       (map #(select-keys (field-values/get-or-create-full-field-values! %)
                                                  [:field_id :human_readable_values :values])
                                     (t2/hydrate (t2/select :model/Field :id [:in (set field-ids)]) :values)
```

I haven't profiled this to see how much it will actually slow down things, but it seems like an obvious performance win.